frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/base/insert_page_command.dart

```diff
@@ -1,34 +1,77 @@
+import 'package:appflowy/generated/locale_keys.g.dart';
+import 'package:appflowy/plugins/database_view/application/database_view_service.dart';
 import 'package:appflowy/plugins/document/presentation/plugins/board/board_node_widget.dart';
 import 'package:appflowy/plugins/document/presentation/plugins/grid/grid_node_widget.dart';
+import 'package:appflowy/workspace/application/app/app_service.dart';
+import 'package:appflowy_backend/protobuf/flowy-database/database_entities.pb.dart';
 import 'package:appflowy_backend/protobuf/flowy-folder/app.pb.dart';
 import 'package:appflowy_backend/protobuf/flowy-folder/view.pb.dart';
 import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:easy_localization/easy_localization.dart';
 
 const String kAppID = 'app_id';
 const String kViewID = 'view_id';
 
 extension InsertPage on EditorState {
-  void insertPage(AppPB appPB, ViewPB viewPB) {
+  void insertPage(AppPB appPB, ViewPB viewPB) async {
     final selection = service.selectionService.currentSelection.value;
     final textNodes =
         service.selectionService.currentSelectedNodes.whereType<TextNode>();
     if (selection == null || textNodes.isEmpty) {
       return;
     }
+
+    // get the database that the view is associated with
+    final DatabasePB? database =
+        await DatabaseViewBackendService(viewId: viewPB.id).openGrid().then(
+              (value) => value.getLeftOrNull(),
+            );
+
+    if (database == null) {
+      throw StateError('''
+The database associated with ${viewPB.id} could not be found while attempting to create a referenced ${viewPB.layout.name}.
+''');
+    }
+
+    final String prefix = referencedBoardPrefix(viewPB.layout);
```

Comment:
https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides#5dc0

Reply:
I'll either remove or add all type qualifiers for this PR.

Reply:
Fair enough.

I've never seen a need for typing variables when the type is easily inferred, also don't remember anything like that from the Dart Effective Guidelines. I see Nathan's argument that just because the compiler can infer it does not mean programmers can. But that might be true for other languages, but the intellisense support for Dart, along with the fact it's a strongly typed language, means we never have to "guess" the type.

I'll keep lookout on that page or any PRs to docs regarding this, it seems there are still some unresolved choices to be made.

Reply:
The [Flutter Style Guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var-and-dynamic) has a similar opinion, but they're all just opinions at the end of the day. 

> This achieves two purposes: it verifies that the type that the compiler would infer matches the type you expect, and it makes the code self-documenting in the case where the type is not obvious (e.g. when calling anything other than a constructor).

Normally, I write the type name for everything, but lately I've been trying to just use final/var. I haven't done enough to say which one is better, so for now I'm just going to focus on landing this PR.

Reply:
cc @LucasXu0 what do you think? I noticed that you don't use full type names for your variables.